### PR TITLE
TST: added windows to CI

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #76 and #90.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality or documentation)

# How Has This Been Tested?

Testing it on CI, since I don't have a windows machine

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by JonathonMSmith** [comment]
Just as an update, I ran the test suite on windows 10 with python 3.10. I did not get the memory allocation error but I did get a couple others. I notice that the WinError32 is happening in one of the python 3.9 jobs here as well.
```
=============================================== short test summary info ===============================================
FAILED pysatMadrigal/tests/test_instruments.py::TestInstruments::test_download[inst_dict15] - ValueError: unable to open url http://cedar.openmadrigal.org/getExperimentsService.py?code=8000&startyear=2001&star...
FAILED pysatMadrigal/tests/test_instruments.py::TestInstruments::test_download[inst_dict18] - TimeoutError: timed out
ERROR pysatMadrigal/tests/test_methods_general.py::TestNetCDFFiles::test_load_netcdf[1] - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\...
ERROR pysatMadrigal/tests/test_methods_general.py::TestNetCDFFiles::test_load_netcdf[2] - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\...
ERROR pysatMadrigal/tests/test_methods_general.py::TestNetCDFFiles::test_load_netcdf[3] - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\...
ERROR pysatMadrigal/tests/test_methods_general.py::TestNetCDFFiles::test_load_netcdf_extra_xarray_coord - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\...
==================== 2 failed, 306 passed, 9 skipped, 355 warnings, 4 errors in 9812.04s (2:43:32) ====================
```

**Comment by aburrell** [comment]
Thanks @JonathonMSmith I know we fixed the WinErr32 before.  @jklenzing do you remember what we did or where it popped up?

**Comment by jklenzing** [comment]
> Thanks @JonathonMSmith I know we fixed the WinErr32 before. @jklenzing do you remember what we did or where it popped up?

I think we had to add a `data.close` statement after reading in the data to close the link to the file. Windows keeps it open for some reason.

**Comment by jklenzing** [comment]
Added the fix for the workflow in #90. Cancelling tests for now since #89 needs to be fixed before this branch will work.

**Comment by aburrell** [comment]
A new error!

```
ERROR pysatMadrigal/tests/test_methods_general.py::TestNetCDFFiles::test_load_netcdf[1] - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpegy6o9b4\\temp0.netCDF4'
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
===== 245 passed, 1 skipped, 331 warnings, 1 error in 8401.79s (2:20:01) ======
    with os.scandir(path) as scandir_it:
NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpegy6o9b4\\temp0.netCDF4'
Error: Process completed with exit code 1.
```

@jklenzing @JonathonMSmith ideas?

**Comment by JonathonMSmith** [comment]
interesting, so it looks like first there is the old WinError 32, and then (... after the testing is finished?) WinError 267. Can we fix the 32 with a manual close? It almost looks like error 267 is happening after the unit tests have been run

**Comment by aburrell** [comment]
> Can we fix the 32 with a manual close?

I added manual closes in multiple places.  Initially it worked, and then it popped up again.  I checked our previous issue for this in pysat and found that we needed to use a general Exception, because Windows will throw SO MANY ERRORS with the tempfiles.

- https://github.com/pysat/pysatMadrigal/pull/88/files#diff-97fd3f94000951561fffcc6460d1ec2bdfecdfa6fe4b1de0893fc0db49fd8f91R506
- https://github.com/pysat/pysatMadrigal/pull/88/files#diff-97fd3f94000951561fffcc6460d1ec2bdfecdfa6fe4b1de0893fc0db49fd8f91R617
- https://github.com/pysat/pysatMadrigal/pull/88/files#diff-97fd3f94000951561fffcc6460d1ec2bdfecdfa6fe4b1de0893fc0db49fd8f91R636

File: pysatMadrigal/tests/test_methods_general.py [hunk 1]
```diff
@@ -502,8 +502,20 @@ def teardown_method(self):
         # Remove the temporary directory and file
         for tfile in self.temp_files:
             if os.path.isfile(tfile):
-                os.remove(tfile)
-        self.data_path.cleanup()
+                if hasattr(tfile, 'close'):
+                    tfile.close()
+
```
Comment (aburrell) [inline pysatMadrigal/tests/test_methods_general.py:507]:
```suggestion
```

This line is never reached.  https://coveralls.io/builds/61025359/source?filename=pysatMadrigal%2Ftests%2Ftest_methods_general.pyhttps://coveralls.io/builds/61025359/source?filename=pysatMadrigal%2Ftests%2Ftest_methods_general.py

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 
 [0.1.X] - 2023-XX-XX
 --------------------
+* Enhancements
+  * Moved the NGDC AE index Instrument from the general Madrigal Pandas
+    instrument to it's own, fixing the Windows memory issue and a problem
+    with duplicated times
```
Comment (jklenzing) [inline CHANGELOG.md:11]:
```suggestion
  * Moved the NGDC AE index Instrument from the general Madrigal Pandas
    instrument to a new one, fixing the Windows memory issue and a problem
    with duplicated times
```

**Review by jklenzing — CHANGES_REQUESTED** [review]

**Review by jklenzing — APPROVED** [review]

**Pull request merged**